client/test/unit/lib/solana/transactions/expandTransaction.test.ts

```diff
@@ -0,0 +1,148 @@
+            add: AssignablePublicKey.fromPublicKey(
+              Keypair.generate().publicKey
+            ),
+          });
+        case 1:
+          return new AccountOperation({
+            clear: new UnitValue(),
+          });
+        case 2:
+          return new AccountOperation({
+            addMany: range(0, randomInt(2, 10)).map(() =>
+              AssignablePublicKey.fromPublicKey(Keypair.generate().publicKey)
+            ),
+          });
+        default:
+          throw new Error('Out of bounds');
+      }
+    });
+
+    const instructionOperations: InstructionOperation[] = range(
+      0,
+      randomInt(2, 10)
+    ).map(() => {
+      switch (randomInt(0, 8)) {
+        case 0:
+          return new InstructionOperation({
+            push: new InstructionData({
+              accounts: range(0, randomInt(2, 10)).map(() =>
+                TransactionAccountMeta.random()
+              ),
+              data: new Uint8Array(randomInt(1, 1000)),
+              program_id: randomInt(0, 10),
+            }),
+          });
+        case 1:
+          return new InstructionOperation({
+            pop: new UnitValue(),
+          });
+        case 2:
+          return new InstructionOperation({
+            addAccount: new AddAccount({
+              index: randomInt(0, 10),
+              account: TransactionAccountMeta.random(),
+            }),
+          });
+        case 3:
+          return new InstructionOperation({
+            addAccounts: new AddAccounts({
+              index: randomInt(0, 10),
+              account: range(0, randomInt(2, 10)).map(() =>
+                TransactionAccountMeta.random()
+              ),
+            }),
+          });
+        case 4:
+          return new InstructionOperation({
+            clearAccounts: randomInt(0, 10),
+          });
+        case 5:
+          return new InstructionOperation({
+            addData: new AddData({
+              index: randomInt(0, 10),
+              data: new Buffer(randomInt(1, 1000)),
+            }),
+          });
+        case 6:
+          return new InstructionOperation({
+            clearData: randomInt(0, 10),
+          });
+        case 7:
+          return new InstructionOperation({
+            clear: new UnitValue(),
+          });
+        default:
+          throw new Error('Out of bounds');
+      }
+    });
+
+    const seed = Math.floor(Math.random() * 100000).toString();
+    const instruction1 = await create(
+      accountOperations,
+      instructionOperations,
+      didPDAKey,
+      await deriveDefaultDOAFromKey(didPDAKey),
+      seed,
+      false,
+      [normalizeSigner(payer), []]
+    );
+
+    console.log(instruction1);
+
+    const instruction2 = await create(
+      accountOperations,
+      instructionOperations,
+      didPDAKey,
+      await deriveDefaultDOAFromKey(didPDAKey),
+      seed,
+      true,
+      [normalizeSigner(payer), []]
+    );
+
+    console.log(instruction2);
```

Comment:
should there be some assertions going on here, or is it sufficient that nothing breaks? Either way I suggest a comment or two

Reply:
I meant to add them